src/libs/ajax/WorkspaceDataService.ts

```diff
@@ -70,6 +85,30 @@ export const WorkspaceData = (signal) => ({
     );
     return res.json();
   },
+  getCapabilities: async (root: string): Promise<Capabilities> => {
+    return fetchWDS(root)('capabilities/v1', _.mergeAll([authOpts(), { signal, method: 'GET' }]))
+      .then(async (response) => {
+        const json = await response.json();
+        const capabilities: Capabilities = {};
+
+        for (const key in json) {
+          if (Object.prototype.hasOwnProperty.call(json, key)) {
+            // Default to false if not boolean
+            const jsonValue = typeof json[key] === 'boolean' ? json[key] : false;
+            capabilities[key as Capability] = jsonValue;
+          }
+        }
+
+        return capabilities;
+      })
+      .catch((error) => {
+        if (error instanceof Response && error.status === 404) {
+          console.log("WDS doesn't support capabilities endpoint"); // eslint-disable-line no-console
+          return { capabilities: false } as Capabilities;
+        }
+        return Promise.reject(error);
```

Comment:
```suggestion
        throw error;
```

Would probably be more idiomatic. The callback here can throw and `catch` will return a Promise rejected with the thrown value.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch#return_value

The documentation for `then` elaborates on the return value: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#return_value

Reply:
Fixed in af036fab9cc4f3c5f80b0df32a4a8199d7a27444

Reply:
Thanks for the references!